src/renderer/entities/transaction/ui/QrCode/QrReader/QrSignatureReader.tsx

```diff
@@ -111,57 +111,41 @@ export const QrSignatureReader = ({
           decodeCallback,
         );
       }
+
       onStart?.();
     } catch {
       throw QR_READER_ERRORS[QrError.DECODE_ERROR];
     }
   };
 
-  const stopScanning = () => {
-    if (streamRef.current) {
-      for (const track of streamRef.current.getVideoTracks()) {
-        track.stop();
-      }
-    }
-
-    controlsRef.current?.stop();
-    bgControlsRef.current?.stop();
+  const handleStopScanning = () => {
+    void stopScanning({ streamRef, controlsRef, bgControlsRef });
   };
 
   useEffect(() => {
     (async () => {
       try {
-        const camerasAmount = await getVideoInputs();
-        scannerRef.current = new BrowserQRCodeReader(undefined, {
-          delayBetweenScanAttempts: 50,
-          delayBetweenScanSuccess: 50,
-        });
-
-        if (!camerasAmount || camerasAmount === 1) {
-          await startScanning();
+        if (!scannerRef.current) {
+          const camerasAmount = await getVideoInputs();
+          scannerRef.current = new BrowserQRCodeReader(undefined, {
+            delayBetweenScanAttempts: 50,
+            delayBetweenScanSuccess: 50,
+          });
+
+          if (!cameraId && (!camerasAmount || camerasAmount === 1)) {
+            await startScanning();
+          }
         }
       } catch (error) {
-        onError?.(error as ErrorObject);
+        {
+          onError?.(error as ErrorObject);
+        }
       }
     })();
 
     return () => {
-      stopScanning();
+      handleStopScanning();
     };
-  }, []);
-
-  useEffect(() => {
-    if (!cameraId) return;
-
-    (async () => {
-      try {
-        controlsRef.current?.stop();
-        bgControlsRef.current?.stop();
-        await startScanning();
-      } catch {
-        onError?.(QR_READER_ERRORS[QrError.BAD_NEW_CAMERA]);
-      }
-    })();
```

Comment:
the main issue was that startScanning() called twice which run 2 usage objects. when successfully scan - closed only one and the second one remain opened. It become worse, when user make operation multiple times